moveit_core/collision_distance_field/src/collision_env_distance_field.cpp

```diff
@@ -942,8 +942,8 @@ DistanceFieldCacheEntryPtr CollisionEnvDistanceField::generateDistanceFieldCache
       for (collision_detection::PosedBodyPointDecompositionVectorPtr& non_group_attached_body_decomposition :
            non_group_attached_body_decompositions)
       {
-        all_points.insert(all_points.end(), non_group_attached_body_decomposition->getCollisionPoints().begin(),
```

Comment:
> Calling getCollisionPoints on non_group_attached_body_decomposition (see here) returns iterators to different vectors

How are they different ? [getCollisionPoints](https://github.com/ros-planning/moveit2/blob/main/moveit_core/collision_distance_field/include/moveit/collision_distance_field/collision_distance_field_types.h#L358) returns a reference to a member variable

Reply:
I guess my error description is not precise. The two calls to `getCollisionPoints` in the line referenced above provide two different copies of the collision points. So, when `begin()` and `end()` are called, it is the beginning and end of two **different** objects. Instead, it needs to be the beginning and end of the **same** object to make the insertion work. See also: http://www.cplusplus.com/forum/beginner/173704/#msg861903.

Reply:
As you can see from the link `getVec` returns `std::vector<double>` whereas `getCollisionPoints` returns `const EigenSTL::vector_Vector3d&` they're not the same

Reply:
Do you mind sharing the stacktrace ?

Reply:
I think you were looking at PosedBodyPointDecomposition instead of PosedBodyPointDecompositionVector. The latter's [getCollisionPoints](https://github.com/ros-planning/moveit2/blob/main/moveit_core/collision_distance_field/include/moveit/collision_distance_field/collision_distance_field_types.h#L451-L459) does not return a reference to a member variable.

Reply:
Oh, now I see, thanks!